dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EventDAO.java

```diff
@@ -68,13 +76,29 @@ public long countAllEventsForOrganization(long organizationId) {
 
     public List<Event> findAllByOrganizationIdsOrEntryIds(Set<Long> organizationIds, Set<Long> entryIds, Integer offset, int limit) {
         int newLimit = Math.min(MAX_LIMIT, limit);
-        if (organizationIds.isEmpty()) {
+
+        CriteriaBuilder cb = currentSession().getCriteriaBuilder();
+        CriteriaQuery<Event> query = criteriaQuery();
+        Root<Event> event = query.from(Event.class);
+
+        if (organizationIds.isEmpty() && entryIds.isEmpty()) {
             return Collections.emptyList();
         }
-        Query<Event> query = this.currentSession().getNamedQuery("io.dockstore.webservice.core.Event.findAllByOrganizationIdsOrEntryIds");
-        query.setParameterList("organizationIDs", organizationIds).setParameter("entryIDs", entryIds)
-                .setFirstResult(offset).setMaxResults(newLimit);
-        return list(query);
+        List<Predicate> list = new ArrayList<>();
+        if (!organizationIds.isEmpty()) {
+            list.add(event.get("organization").in(organizationIds));
```

Comment:
Does this work correctly? Doesn't `event.get("organization")` return an Organization object? And it's trying to see if it's in a list of Long?

I know this gets converted into SQL, so maybe it ends up automagically working, that it knows to get the id of the organization, and you have passing tests, but still, just want to be sure.

Reply:
Yeah, I can get events from both nf-core and seven bridges interweaved on the homepage. 
There must be some automagic